rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/impl/AsyncResponseImpl.java

```diff
@@ -302,6 +302,11 @@ public synchronized void handleTimeout() {
     private void initContinuation() {
         ContinuationProvider provider =
             (ContinuationProvider)inMessage.get(ContinuationProvider.class.getName());
+        if (provider == null) {
+            throw new IllegalArgumentException(
```

Comment:
Very minor, I would probably go with `IllegalStateException` (seems to fit better to this particular issue) but `IllegalArgumentException` is also fine.

Reply:
The filters are a kind of argument, aren't they? When defining a filter in `web.xml` or via `@WebFilter`… So yes, both are probably fine.

Anyway, the important thing here is that users see an exception with a somewhat meaningful message. Without this fix, if you use e.g. Apache Shiro `shiro-servlet:1.4.0`, you can only see a `NullPointerException` with a CXF stack trace, which is not helpful at all. See the stack trace here: https://github.com/bmhm/shiro-openliberty-npe and the issue for Apache SHIRO here: https://issues.apache.org/jira/projects/SHIRO/issues/SHIRO-735.